primitives/session/src/lib.rs

```diff
@@ -130,3 +130,15 @@ where
 
 	Ok(())
 }
+
+/// A trait to be notified when the session changes.
+/// This is in contrast to `SessionHandler` which handles events for specific keys.
+#[impl_trait_for_tuples::impl_for_tuples(30)]
+pub trait SessionChangeListener {
+	fn on_session_change(session_index: SessionIndex);
+}
+
+/// A trait get the current session info
```

Comment:
An idiom that I was thought years ago and sometimes follow is:

```rust
trait SomeOPerationName { fn some_operation_name() -> () }
```

but ofc that's just an opinion.

Reply:
Yes, I was hoping that this trait will expand later to provide a lot more info, but happy to keep it specific for now.